dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/EventQueryParams.java

```diff
@@ -500,6 +503,33 @@ public boolean orgUnitFieldIsValid()
             return true;
         }
 
+        if ( program != null )
+        {
+            return validateProgramHasOrgUnitField( program );
+        }
+
+        if ( !itemProgramIndicators.isEmpty() )
+        {
+            boolean[] isValidated = new boolean[itemProgramIndicators.size()];
+
+            for ( int i = 0; i < itemProgramIndicators.size(); i++ )
+            {
+                ProgramIndicator itemProgramIndicator = itemProgramIndicators.get( i );
+                Program program = itemProgramIndicator.getProgram();
+
+                isValidated[i] = validateProgramHasOrgUnitField( program );
+
+            }
+            // if at least one Program Indicator doesn't validate, fail the validation
+            return !Booleans.contains( isValidated, false );
```

Comment:
Since "if at least one Program Indicator doesn't validate, we fail the validation", could instead of checking all program indicators just break when we find one program indicator that does not validate?

(It is needless to check all if we fail at the first false).

Reply:
yes, absolutely. Is the logic ok, though? I kind of made it up myself. For instance, if all the PI fails shall we considering falling back to the reporting unit?

Reply:
Simpliefied the logic  using `stream.anyMatch`

Reply:
I think the logic makes sense.